src/es/ravenmanga/src/eu/kanade/tachiyomi/extension/es/ravenmanga/RavenManga.kt

```diff
@@ -0,0 +1,186 @@
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import okhttp3.Headers
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import uy.kohesive.injekt.injectLazy
+import java.io.IOException
+import java.lang.IllegalArgumentException
+import java.util.Calendar
+
+class RavenManga : ParsedHttpSource() {
+
+    override val name = "RavenManga"
+
+    override val baseUrl = "https://ravenseries.lat"
+
+    override val lang = "es"
+
+    override val supportsLatest = true
+
+    override val client: OkHttpClient = network.client.newBuilder()
+        .rateLimitHost(baseUrl.toHttpUrl(), 2)
+        .build()
+
+    override fun headersBuilder(): Headers.Builder = Headers.Builder()
+        .add("Referer", baseUrl)
+
+    private val json: Json by injectLazy()
+
+    override fun popularMangaRequest(page: Int): Request = GET(baseUrl, headers)
+
+    override fun popularMangaSelector(): String = "div#div-diario figure, div#div-semanal figure, div#div-mensual figure"
+
+    override fun popularMangaNextPageSelector(): String? = null
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val document = response.asJsoup()
+        val elements = document.select(popularMangaSelector())
+
+        val mangas = elements.map { element ->
+            SManga.create().apply {
+                thumbnail_url = element.selectFirst("img")!!.attr("abs:src")
+                title = element.selectFirst("figcaption")!!.text()
+                setUrlWithoutDomain(element.selectFirst("a")!!.attr("href"))
+            }
+        }.distinctBy { it.url }
+
+        return MangasPage(mangas, false)
+    }
+
+    override fun popularMangaFromElement(element: Element): SManga = throw UnsupportedOperationException("Not used!")
+
+    override fun latestUpdatesRequest(page: Int): Request = GET(baseUrl, headers)
+
+    override fun latestUpdatesSelector(): String = "section.flex > div.grid > figure"
+
+    override fun latestUpdatesNextPageSelector(): String? = null
+
+    override fun latestUpdatesFromElement(element: Element): SManga = SManga.create().apply {
+        thumbnail_url = element.selectFirst("img")!!.attr("abs:src")
+        title = element.selectFirst("figcaption")!!.text()
+        setUrlWithoutDomain(element.selectFirst("a")!!.attr("href"))
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        if (query.isNotEmpty()) {
+            if (query.length > 1) return GET("$baseUrl/comics#$query", headers)
+            throw IOException("La búsqueda debe tener al menos 2 caracteres")
```

Comment:
Can use regular `Exception` here.

```suggestion
            throw Exception("La búsqueda debe tener al menos 2 caracteres")
```